### PR TITLE
Prettier eslint

### PR DESCRIPTION
**Pull request by bensalilijames**
This supercedes #5497: instead of using `standard`, this PR switches to `eslint` with `prettier` and `eslint-config-airbnb`.

I've only run `eslint` on the `examples` folder as @timneutkens suggested - I think this is a good first step and validates that the tooling works!

There are currently `1728 problems (1645 errors, 83 warnings)` in the `examples` folder. Rather than fix these all by hand, I've added `eslint` to a pre-commit step, so that if they get changed, those problems will need to be fixed before committing. That way we can (forcefully!) incrementally fix the problems.

I've also opted not to add `eslint` to the `lint` script in `package.json` yet - otherwise the CI lint step will fall over until all the above problems are fixed.

`standard` should no longer run on the `examples` folder.

Looking forward to hearing your thoughts! 🙂

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
+  extends: ["airbnb", "prettier", "prettier/react"],
+  parser: "babel-eslint",
+  plugins: ["prettier"],
+  rules: {
+    "prettier/prettier": ["error"],
+    // Allow JSX in .js file extension as well as default .jsx
+    "react/jsx-filename-extension": ["error", { extensions: [".js", ".jsx"] }],
+    // next.js automatically has React in scope, so turn off requiring React
+    "react/react-in-jsx-scope": "off"
```
Comment (mherodev) [inline .eslintrc.js]: Is this true for projects that use Next.js? e.g. the project I'm currently working on? 

I'd love to remove the redundancy 💃

Comment (timneutkens) [reply .eslintrc.js]: `import React from 'react'` is automatically injected when using JSX, https://github.com/vslinko/babel-plugin-react-require is included

**Comment by timneutkens** [comment]
Thanks so much for this PR!

Sorry for changing my mind, but I think the most gradual way we can adopt this is implementing your previous PR only for the examples directory, so keep using standardjs and then add prettier. It's pretty much impossible to adopt this with `1728 problems (1645 errors, 83 warnings)` so gradual adoption seems like a better idea 👌 I hope you agree 🙏

**Comment by benhjames** [comment]
Sounds like a good idea - let's do that! When I find a moment I'll modify the old PR. 👍

**Comment by timneutkens** [comment]
Awesome! Thanks so much Ben 🙏

**Pull request closed without merging**